ui/desktop/src/components/settings/extensions/modal/ExtensionModal.test.tsx

```diff
@@ -0,0 +1,89 @@
+import { describe, it, expect, vi } from 'vitest';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ExtensionModal from './ExtensionModal';
+import { ExtensionFormData } from '../utils';
+
+describe('ExtensionModal', () => {
+  it('creates a http_streamable extension', async () => {
+    const user = userEvent.setup();
+    const mockOnSubmit = vi.fn();
+    const mockOnClose = vi.fn();
+
+    const initialData: ExtensionFormData = {
+      name: '',
+      description: '',
+      type: 'stdio', // Default type
+      cmd: '',
+      endpoint: '',
+      enabled: true,
+      timeout: 300,
+      envVars: [],
+      headers: [],
+    };
+
+    render(
+      <ExtensionModal
+        title="Add custom extension"
+        initialData={initialData}
+        onClose={mockOnClose}
+        onSubmit={mockOnSubmit}
+        submitLabel="Add Extension"
+        modalType="add"
+      />
+    );
+
+    const nameInput = screen.getByPlaceholderText('Enter extension name...');
+    const submitButton = screen.getByTestId('extension-submit-btn');
+
+    await user.type(nameInput, 'Test MCP');
+
+    const typeSelect = screen.getByRole('combobox');
+    await user.click(typeSelect);
+
+    const httpOption = screen.getByText('Streamable HTTP');
+    await user.click(httpOption);
+
+    await waitFor(() => {
+      expect(screen.getByText('Request Headers')).toBeInTheDocument();
+    });
+
+    const endpointInput = screen.getByPlaceholderText('Enter endpoint URL...');
+    await user.type(endpointInput, 'https://foo.bar.com/mcp/');
+
+    const descriptionInput = screen.getByPlaceholderText('Optional description...');
+    await user.type(descriptionInput, 'Test MCP extension');
+
+    const headerNameInput = screen.getByPlaceholderText('Header name');
+    const headerValueInput = screen
+      .getAllByPlaceholderText('Value')
+      .find(
+        (input) =>
+          input.closest('div')?.textContent?.includes('Request Headers') ||
+          input.parentElement?.parentElement?.textContent?.includes('Request Headers')
+      );
+
+    await user.type(headerNameInput, 'Authorization');
+    if (headerValueInput) {
+      await user.type(headerValueInput, 'Bearer abc123');
+    }
+
+    await user.click(submitButton);
+
+    await waitFor(() => {
+      expect(mockOnSubmit).toHaveBeenCalled();
+    });
+
+    const submittedData = mockOnSubmit.mock.calls[0][0];
+
+    expect(submittedData.name).toBe('Test MCP');
+    expect(submittedData.type).toBe('streamable_http');
+    expect(submittedData.endpoint).toBe('https://foo.bar.com/mcp/');
+    expect(submittedData.description).toBe('Test MCP extension');
+    expect(submittedData.timeout).toBe(300);
+    expect(submittedData.headers).toHaveLength(1);
+    expect(submittedData.headers).toEqual([
+      { key: 'Authorization', value: 'Bearer abc123', isEdited: true },
+    ]);
+  });
+});
```

Comment:
this test doesn't seem to test the breakage, it just seems to test that you can fill in the form. I think as is, it is only going to break if we change our form or texts.